eth/filters/api.go

```diff
@@ -203,15 +201,14 @@ func (s *PublicFilterAPI) NewPendingTransactionFilter() (string, error) {
 
 // newLogFilter creates a new log filter.
 func (s *PublicFilterAPI) newLogFilter(earliest, latest int64, addresses []common.Address, topics [][]common.Hash, callback func(log *vm.Log, removed bool)) (int, error) {
-	s.logMu.Lock()
-	defer s.logMu.Unlock()
-
 	filter := New(s.chainDb)
-	id, err := s.filterManager.Add(filter, LogFilter)
+	id, err := s.filterManager.Add(filter, LogFilter) // use inside logMu deadlocks filterLoop()
```

Comment:
Could you explain?
> use inside logMu deadlocks filterLoop()

Locking after the Add could lead to missing logs when logs are raised and the new filter is not added to the logQueue. Afaik a deadlock is not feasible but it could stall the filterloop for a brief time when the function has not returned and new logs are generated that match the filter criteria.